### ISSUE TITLE
[Improvement] Better error message when in detached head

### ISSUE THREAD

**Issue by Evangelink** [GitTools/GitVersion#2753]
**Is your improvement request related to a problem? Please describe.**

When re-building a commit in history, I often fail to remember to comment the `GitVersion.MsBuild` package reference from my project. Which results in one failure per project with the following message:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3073	The command ""C:\Users\XXX\.nuget\packages\gitversion.msbuild\5.6.10\tools\net48/gitversion.exe" "C:\Src\YYY" -output file -outputfile obj\gitversion.json" exited with code 1.	YYY	C:\Users\XXX\.nuget\packages\gitversion.msbuild\5.6.10\tools\GitVersion.MsBuild.targets	9	
```

This error message is pretty generic and could (maybe?) be improved for the case of detached head with a more helpful message.

## Detailed Description

Either return a different exit code and document it and/or fail with a message that would be displayed. I haven't done msbuild tasks for quite some time and so I don't really remember what's possible and how to do so.

## Context

Relates to #2697

**Comment by asbjornu** [comment]
Improvements in error messages are always welcome! A PR would be highly appreciated. 🙏🏼

**Comment by Evangelink** [comment]
I will try to provide something but if you have some spare time for a couple of advices, they would be welcome :)

**Comment by asbjornu** [comment]
If you are able to reproduce the problem with the command line, we can see whether the exception thrown is something sensible and start from there.

**Comment by Evangelink** [comment]
Just for the sake of grouping the info together, here are the steps to reproduce the issue:
- Install GitVersion on a project
- Make some commit
- Checkout previous commit
- Rebuild solution
- Build fail with error saying gitversion.exe failed

**Comment by asbjornu** [comment]
A PR that improves the error message of detached heads is still very welcome. :)

**Comment by Evangelink** [comment]
Hey @asbjornu! I am really sorry about all that delay. I had a pretty busy end of year, I will try to give it a go during the week-end.

**Comment by asbjornu** [comment]
No problem, @Evangelink. Take we're in no rush. :)

**Comment by asbjornu** [comment]
Fixed by #3338.

**Comment by nenadvicentic** [comment]
I just got the same error again with 5.12.0:

```shell
  ERROR [07/31/23 12:27:11:55] An unexpected error occurred:
[12:27:11]GitVersion.BugException: GitVersion has a bug, your HEAD has moved after repo normalisation.
[12:27:11]
[12:27:11]To disable this error set an environmental variable called IGNORE_NORMALISATION_GIT_HEAD_MOVE to 1
[12:27:11]
[12:27:11]Please run `git log --graph --format="%h %cr %d" --decorate --date=relative --all --remotes=* -n 100` and submit it along with your build log (with personal info removed) in a new issue at https://github.com/GitTools/GitVersion
[12:27:11]   at GitVersion.GitPreparer.NormalizeGitDirectory(Boolean noFetch, String currentBranchName, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 192
[12:27:11]   at GitVersion.GitPreparer.NormalizeGitDirectory(String targetBranch, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 134
[12:27:11]   at GitVersion.GitPreparer.PrepareInternal(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 69
[12:27:11]   at GitVersion.GitPreparer.Prepare() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 48
[12:27:11]   at GitVersion.GitVersionCalculateTool.CalculateVersionVariables() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitVersionCalculateTool.cs:line 43
[12:27:11]   at GitVersion.GitVersionExecutor.RunGitVersionTool(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.App\GitVersionExecutor.cs:line 66
[12:27:11]Process exited with code 1
```

When ` IGNORE_NORMALISATION_GIT_HEAD_MOVE to 1` is added, as recommended here, GitVersion incorrectly picks latest commit and calculates wrong version number of the build.
